detection_rules/beats.py

```diff
@@ -286,5 +286,5 @@ def get_schema_from_kql(tree: kql.ast.BaseNode, beats: list, version: str = None
 
 def parse_beats_from_index(index: Optional[list]) -> List[str]:
     indexes = index or []
-    beat_types = [index.split("-")[0] for index in indexes if "beat-*" in index]
+    beat_types = [index.split(':', 1)[-1].split("-")[0] for index in indexes if "beat-*" in index]
     return beat_types
```

Comment:
nit:

for readability, early exiting, etc., what about something like this.

```
def parse_beats_from_index(indexes: Optional[List[str]]) -> List[str]:
    if not indexes:
        return []
    
    beat_types = []
    for index in indexes:
        if "beat-*" in index:
            split_index = index.split(':', 1)
            if len(split_index) > 1:
                beat_type = split_index[1].split("-")[0]
                beat_types.append(beat_type)
    return beat_types
```

Reply:
Early exiting is still caught by `if "beat-*" in index`. Happy to update to be similar to this for readability :+1:

Reply:
Good point. It's just a nit.

Reply:
+1, I think it might be better to unpack it for readability. `index.split(':', 1)[-1].split("-")[0]` takes a second to decipher 😅

Reply:
Updated